File: js_relay/check_balance.js
const { ethers } = require("ethers");
require('dotenv').config({ path: '/Users/boliu/darwin-workspace/project-darwin/.env' });

const RPC_URL = process.env.BASE_SEPOLIA_RPC || "https://sepolia.base.org";
const OPERATOR_ADDRESS = process.env.OPERATOR_ADDRESS;

async function main() {
  const provider = new ethers.JsonRpcProvider(RPC_URL);
  const balance = await provider.getBalance(OPERATOR_ADDRESS);
  console.log(`Address: ${OPERATOR_ADDRESS}`);
  console.log(`Balance: ${ethers.formatEther(balance)} ETH`);
}

main();

File: check_balance.py
from web3 import Web3
import os

rpc = "https://sepolia.base.org"
w3 = Web3(Web3.HTTPProvider(rpc))
address = "0xE9A11d3304b549c09403865E8220a05951C575A6"
balance_wei = w3.eth.get_balance(address)
balance_eth = w3.from_wei(balance_wei, 'ether')

print(f"Address: {address}")
print(f"Balance: {balance_eth} ETH")
